### PR TITLE
Issue #2460 - Add support for multiple DBs for github

### PR DESCRIPTION
**Pull request by ksy36**

### CODE REVIEW

File: bugbug/model.py [hunk 1]
```diff
@@ -756,12 +756,3 @@ def items_gen(self, classes):
 class IssueModel(Model):
     def __init__(self, lemmatization=False):
         Model.__init__(self, lemmatization)
-        self.training_dbs = [github.GITHUB_ISSUES_DB]
-
-    def items_gen(self, classes):
-        for issue in github.get_issues():
-            issue_number = issue["number"]
-            if issue_number not in classes:
-                continue
```
Comment (ksy36) [inline bugbug/model.py:765]: Moved `self.training_dbs` and `items_gen` to needsdiagnosis model

File: bugbug/github_dbs_config.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+config = {
+    "web-bugs": {
+        "path": "data/github_issues.json",
+        "url": "https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_issues.latest/artifacts/public/github_issues.json.zst",
+        "version": 1,
+    }
+}
```
Comment (ksy36) [inline bugbug/github_dbs_config.py:12]: created this config with a key as repository name, so we could add multiple DBs

File: requirements.txt [hunk 3]
```diff
@@ -18,7 +18,7 @@ python-dateutil==2.8.2
 python-hglib==2.6.2
 ratelimit==2.2.1
 requests==2.26.0
-rs_parsepatch==0.3.5
+rs_parsepatch==0.3.6
```
Comment (marco-c) [inline requirements.txt:21]: Unrelated change (maybe leftover from merge?)

File: scripts/github_issue_retriever.py [hunk 4]
```diff
@@ -7,98 +7,111 @@
...
                     "Deleting public issues that were updated and saving updates"
                 )
-                github.delete_issues(lambda issue: issue["id"] in updated_ids)
-                db.append(github.GITHUB_ISSUES_DB, updated_issues)
+                self.github.delete_issues(lambda issue: issue["id"] in updated_ids)
+                db.append(self.github.db_path, updated_issues)
+
+        zstd_compress(self.github.db_path)
+
+    def replace_with_private(
```
Comment (marco-c) [inline scripts/github_issue_retriever.py:88]: Could you keep this function where it was, to minimize the diff?

File: bugbug/model.py (hunk 1 above)
Comment (marco-c) [reply bugbug/model.py:765]: Could we instead keep them here, adding `owner` and `repo` as parameters to `IssueModel` so it can create a `GitHub` instance and store it in `self.github`?

File: bugbug/github_dbs_config.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (marco-c) [inline bugbug/github_dbs_config.py:1]: I'd like to find another way than having a module listing all GitHub DBs.
What if we remove this module and call `db.register` in the `GitHub` constructor? The path for the DB could be programmatically built like this:
```
https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_{owner}_{repo}_issues.latest/artifacts/public/github_{owner}_{repo}_issues.json.zst
```

Comment (ksy36) [reply bugbug/github_dbs_config.py:1]: Sounds good, I'll move it to the same file where GitHub class is, but probably not inside the constructor since `db.register` will be called each time Github instance is created?

Comment (ksy36) [reply bugbug/github_dbs_config.py:1]: I could probably check whether it's registered with `db.is_registered()` and not register it again. However there is also db version, which could be different for every DB, so it seems that some sort of config is still needed 🤔

Comment (marco-c) [reply bugbug/github_dbs_config.py:1]: The version could be a constant in the `GitHub` module, since the version only changes when we make changes to the way we store data for GitHub repos, so it sill be the same for any GitHub DB.

File: requirements.txt (hunk 3 above)
Comment (ksy36) [reply requirements.txt]: Right, thanks, I've reverted the change

File: bugbug/github_dbs_config.py (hunk 5 above)
Comment (ksy36) [reply bugbug/github_dbs_config.py]: Thanks! I've made it a constant value.

File: bugbug/model.py (hunk 1 above)
Comment (ksy36) [reply bugbug/model.py:765]: Yeah, added it back

File: scripts/github_issue_retriever.py (hunk 4 above)
Comment (ksy36) [reply scripts/github_issue_retriever.py]: I've moved it inside the Retriever class because it's using its github instance in `self.github.fetch_issue_by_number`

**Comment by ksy36** [comment]
I've made the changes, could you please take another look @marco-c?

**Comment by marco-c** [comment]
@ksy36 thanks, LGTM!
We'll also need to update the paths and URLs in the pipeline definition file at infra/data-pipeline.yml, that is https://github.com/mozilla/bugbug/blob/028bcf19e7f4b941c9e0cca14f78968e99d25886/infra/data-pipeline.yml#L252-L257 and https://github.com/mozilla/bugbug/blob/028bcf19e7f4b941c9e0cca14f78968e99d25886/infra/data-pipeline.yml#L266-L267.

**Comment by ksy36** [comment]
Oops, I've missed that! Will change it as well :)

**Review by marco-c — APPROVED** [review]
Thanks! I'll try to release this today so we can check if everything is working as expected.

**Comment by marco-c** [comment]
CI is misbehaving, I'll contact the Taskcluster team to figure it out.

**Comment by ksy36** [comment]
Thanks!
